packages/nx-plugin/src/generators/plugin-lint-checks/generator.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  addDependenciesToPackageJson,
+  logger,
+  ProjectConfiguration,
+  readJson,
+  readProjectConfiguration,
+  Tree,
+  updateJson,
+  updateProjectConfiguration,
+  writeJson,
+} from '@nrwl/devkit';
+
+import type { Linter as ESLint } from 'eslint';
+
+import { Schema as EsLintExecutorOptions } from '@nrwl/linter/src/executors/eslint/schema';
+
+import { jsoncEslintParserVersion } from '../../utils/versions';
+import { PluginLintChecksGeneratorSchema } from './schema';
+import { NX_PREFIX } from 'nx/src/utils/logger';
+
+export default async function pluginLintCheckGenerator(
+  host: Tree,
+  options: PluginLintChecksGeneratorSchema
+) {
+  const project = readProjectConfiguration(host, options.projectName);
+
+  // This rule is eslint **only**
+  if (projectIsEsLintEnabled(project)) {
+    updateRootEslintConfig(host);
+    updateProjectEslintConfig(host, project);
+    updateProjectTarget(host, options);
+
+    // Project is setup for vscode
+    if (host.exists('.vscode')) {
+      setupVsCodeLintingForJsonFiles(host);
+    }
+  } else {
+    logger.error(
+      `${NX_PREFIX} plugin lint checks can only be added to plugins which use eslint for linting`
+    );
+  }
+}
+
+function updateProjectTarget(
+  host: Tree,
+  options: PluginLintChecksGeneratorSchema
+) {
+  const project = readProjectConfiguration(host, options.projectName);
+  project.targets ??= {};
+  for (const [target, configuration] of Object.entries(project.targets)) {
+    if (configuration.executor === '@nrwl/linter:eslint') {
+      const opts: EsLintExecutorOptions = configuration.options;
+      opts.lintFilePatterns ??= [];
+      opts.lintFilePatterns.push(
+        `${project.root}/generators.json`,
+        `${project.root}/package.json`,
+        `${project.root}/executors.json`
+      );
+      project.targets[target].options = opts;
+    }
+  }
+  updateProjectConfiguration(host, options.projectName, project);
+}
+
+function updateProjectEslintConfig(host: Tree, options: ProjectConfiguration) {
+  // Update the project level lint configuration to specify
+  // the plugin schema rule for generated files
+  updateJson(
+    host,
+    `${options.root}/.eslintrc.json`,
+    (eslintConfig: ESLint.Config) => {
+      eslintConfig.overrides ??= [];
+      eslintConfig.overrides.push({
+        files: ['executors.json', 'package.json', 'generators.json'],
+        parser: 'jsonc-eslint-parser',
```

Comment:
I don't think so, but I'm still debating the root config portion. Probably only going to add it if .vscode folder exists, since its just there for vscode support.